camayoc/tests/qpc/api/v1/conftest.py

```diff
@@ -176,83 +157,93 @@ def run_all_scans(vcenter_client, session_cleanup):
     and one scan will be run against the three sources each created with
     their own credential.
     """
+    cleanup = []
     config = get_config()
-    creds = config['credentials']
-    scans = config.get('qpc', {}).get('scans', [])
-    if scans == []:
+
+    config_scans = config.get('qpc', {}).get('scans', [])
+    if not config_scans:
         # if no scans are defined, no need to go any further
         return
+
+    config_creds = {
+        credential['name']: credential
+        for credential in config.get('credentials', [])
+    }
     inventory = {
         machine['hostname']: machine for machine in config['inventory']
     }
     vcenter_inventory = {
         machine['hostname']: machine for machine in config['inventory']
         if machine.get('hypervisor') == 'vcenter'
     }
-    if not creds or not inventory:
+
+    if not config_creds or not inventory:
         raise ValueError(
             'Make sure to have credentials and inventory'
             ' items in the config file'
         )
-    vcenter_hostnames = list(vcenter_inventory.keys())
-    cred_ids = {}
+
+    credential_ids = {}
     source_ids = {}
-    for cred in creds:
-        # create creds on server
-        # create dict that associates the name of the cred to the cred id
-        cred_ids[cred['name']] = create_cred(cred, session_cleanup)
-    for hostname, source in inventory.items():
-        # create sources on server, and keep track of source ids for each scan
-        # name of cred to cred id on server
-        source_ids[hostname], expected_products = create_source(
-            source,
-            cred_ids,
-            session_cleanup
+    expected_products = {}
+    for scan in config_scans:
+        scan_vms = {
+            vm.name: vm for vm in get_vcenter_vms(vcenter_client)
+            if (vm.name in vcenter_inventory) and (vm.name in scan['sources'])
+        }
+
+        power_on_vms(scan_vms.values())
+
+        for source_name in scan['sources']:
+            if source_name not in source_ids:
+                machine = inventory[source_name]
+                for credential_name in machine['credentials']:
+                    if credential_name not in credential_ids:
+                        credential = config_creds[credential_name].copy()
+                        credential.pop('rho', None)
+                        credential['cred_type'] = credential.pop('type')
+                        credential['ssh_keyfile'] = credential.pop(
```

Comment:
Maybe something to put on the backlog: make the YAML config file naming scheme match the API instead of having to do this munging?

Reply:
I think they differ between CLI and API so we would change were we need to munge

Reply:
I've checked and it matches the cli option: `qpc cred add --name <name> --username <username> --sshkeyfile <sshkeyfile>`. Changing it will require changing CLI stuff to patch the field name so I don't think it worth the time.

Reply:
good call